sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs

```diff
@@ -493,5 +477,80 @@ internal static BlobOpenReadOptions ToBlobOpenReadOptions(this DataLakeOpenReadO
                 Position = options.Position
             };
         }
+
+        //TODO
+        internal static PathSegment ToPathSegment(this ResponseWithHeaders<PathList, FileSystemListPathsHeaders> response)
+            => new PathSegment
+            {
+                Continuation = response.Headers.Continuation,
+                Paths = response.Value.ToPathItems()
+            };
+
+        internal static IEnumerable<PathItem> ToPathItems(this PathList pathList)
+        {
+            if (pathList == null)
+            {
+                return null;
+            }
+
+            return pathList.Paths.Select(path => path.ToPathItem());
+        }
+
+        internal static PathItem ToPathItem(this Path path)
+        {
+            if (path == null)
+            {
+                return null;
+            }
+
+            return new PathItem
+            {
+                Name = path.Name,
+                IsDirectory = path.IsDirectory != null && bool.Parse(path.IsDirectory),
+                LastModified = path.LastModified.GetValueOrDefault(),
+                ETag = new ETag(path.ETag),
+                ContentLength = path.ContentLength == null ? 0 : long.Parse(path.ContentLength, CultureInfo.InvariantCulture),
```

Comment:
can't we generate long?

Reply:
Generator is ignoring long type here for unknown reasons.